skrub/_drop_uninformative.py

```diff
@@ -0,0 +1,134 @@
+
+from . import _dataframe as sbd
+from ._on_each_column import SingleColumnTransformer
+
+__all__ = ["DropUninformative"]
+
+
+class DropUninformative(SingleColumnTransformer):
+    """Drop column if it is found to be uninformative according to various criteria.
+
+    A column is considered to be "uninformative" if one or more of the following
+    issues are found:
+    - The fraction of missing values is larger than a certain fraction (by default,
+    all values must be null for the column to be dropped).
+    - The column includes only one unique value (the column is constant). Missing
+    values are considered a separate value.
+    - The number of unique values in the column is equal to the length of the column.
+
+    Parameters
+    ----------
+    drop_if_constant: bool, default=False
+        If True, drop the column if it contains only one unique value. Missing values
+        count as one additional distinct value.
+    drop_if_id: bool, default=False
+        If True, drop the column if all values are distinct. Missing values count as
+        one additional distinct value. Numeric columns are never dropped.
+    null_fraction_threshold: float or None, default=1.0
+        Drop columns with a fraction of missing values larger than threshold. If None,
+        keep the column even if all its values are missing.
+
+    """
+
+    def __init__(
+        self,
+        drop_if_constant=True,
+        drop_if_id=False,
+        null_fraction_threshold=1.0,
+    ):
+        self.drop_if_constant = drop_if_constant
+        self.drop_if_id = drop_if_id
+        self.null_fraction_threshold = null_fraction_threshold
+
+    def _check_params(self):
+        if not isinstance(self.drop_if_constant, bool):
+            raise TypeError(
+                f"drop_if_constant must be boolean, found {self.drop_if_constant}."
+            )
+        if not isinstance(self.drop_if_id, bool):
+            raise TypeError(f"drop_if_id must be boolean, found {self.drop_if_id}.")
+
+        if self.null_fraction_threshold is not None:
+            if (
+                not isinstance(self.null_fraction_threshold, numbers.Number)
+                or not 0.0 <= self.null_fraction_threshold <= 1.0
+            ):
+                raise ValueError(
+                    f"Threshold {self.null_fraction_threshold} is invalid. Threshold"
+                    " should be a number in the range [0, 1], or None."
+                )
+
+    def _drop_if_too_many_nulls(self, column):
+        if self.null_fraction_threshold == 1.0:
+            return sbd.is_all_null(column)
+        # No nulls found, or no threshold
+        if self._null_count == 0 or self.null_fraction_threshold is None:
+            return False
+        return self._null_count / len(column) > self.null_fraction_threshold
+
+    def _drop_if_constant(self, column):
+        if self.drop_if_constant:
+            if (sbd.n_unique(column) == 1) and (not sbd.has_nulls(column)):
```

Comment:
```suggestion
            if (sbd.n_unique(column) == 1) and (self._null_count == 0):
```

I guess we could avoid re-counting nulls like this?